cli/azd/pkg/tools/azcli/container_registry.go

```diff
@@ -85,6 +87,27 @@ func (crs *containerRegistryService) GetContainerRegistries(
 }
 
 func (crs *containerRegistryService) Login(ctx context.Context, subscriptionId string, loginServer string) error {
+	dockerCreds, err := crs.Credentials(ctx, subscriptionId, loginServer)
+	if err != nil {
+		return err
+	}
+
+	err = crs.docker.Login(ctx, dockerCreds.LoginServer, dockerCreds.Username, dockerCreds.Password)
+	if err != nil {
+		return fmt.Errorf(
+			"failed logging into docker registry %s: %w",
+			loginServer,
+			err)
+	}
+
+	return nil
```

Comment:
Can we add some new unit tests to validate the behavior of the split/add of retrieving credentials from the actual login process?

Reply:
Yup - happy to do this - wanted to get a read from the team on how they felt about this before writing the tests.  Should have made this a draft, :-/.

Reply:
I haven't thought deeply enough, for me it seems more important to:
1. Validate AAD token exchange against a real protocol
2. Validate admin user login against a real protocol

I didn't spot anything too non-trivial that I'd would want additional unit testing at the cost of dev time.

Here's the current code coverage if it helps:
<img width="781" alt="image" src="https://github.com/Azure/azure-dev/assets/2322434/0658c087-f61e-402e-a8ac-0afd21d220f7">